lib/pluginmanager/unpack.rb

```diff
@@ -0,0 +1,47 @@
+require "bootstrap/util/compress"
+require "fileutils"
+
+class LogStash::PluginManager::Unpack < LogStash::PluginManager::Command
+
+  option "--tgz", :flag, "unpack a packaged tar.gz file", :default => !LogStash::Environment.windows?
+  option "--zip", :flag, "unpack a packaged  zip file", :default => LogStash::Environment.windows?
+  option "--force", :flag, "Overwrite previously generated plugins package", :default => false
+
+  parameter "file", "the package file name", :attribute_name => :package_file, :required => true
+
+  def execute
+    puts("Unpacking #{package_file}")
+
+    FileUtils.rm_rf(LogStash::Environment::CACHE_PATH) if force?
+    validate_cache_location
+    unpack(package_file)
+
+    puts("Unpacked at #{LogStash::Environment::CACHE_PATH}")
+    puts("the unpacked plugins can now be installed in local-only mode using bin/plugin install --local [plugin name]")
+  end
+
+  private
+
+  def validate_cache_location
+    location = LogStash::Environment::CACHE_PATH
+    if File.exist?(location)
+      puts("Directory #{location} is going to be overwritten, do you want to continue? (Y/N)")
+      override = ( "y" == STDIN.gets.strip.downcase ? true : false)
+      if override
+        FileUtils.rm_rf(location)
+      else
+        puts("Unpack cancelled: file #{location} already exists, please delete or move it")
+        exit
+      end
+    end
+  end
```

Comment:
I did that not to have the constant everywhere, I think there is nothing wrong with it.